packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/BaseSensorService.kt

```diff
@@ -1,26 +1,46 @@
 // Copyright 2015-present 650 Industries. All rights reserved.
 package expo.modules.sensors.services
 
+import android.Manifest
 import android.content.Context
+import android.content.pm.PackageManager
 import android.hardware.Sensor
 import android.hardware.SensorEventListener2
 import android.hardware.SensorManager
+import android.os.Build
 
 abstract class BaseSensorService internal constructor(reactContext: Context?) : BaseService(reactContext!!), SensorEventListener2 {
   private var mSensor: Sensor? = null
   private val mSensorManager: SensorManager = context.getSystemService(Context.SENSOR_SERVICE) as SensorManager
+  private val samplingPeriodUs: Int
+    get() = if (hasHighSamplingRateSensorsPermission()) SensorManager.SENSOR_DELAY_FASTEST else SensorManager.SENSOR_DELAY_NORMAL
 
   // Abstract methods that subclasses should implement
   abstract val sensorType: Int
 
   // Public API
   protected fun startObserving() {
     if (mSensorManager.getDefaultSensor(sensorType).also { mSensor = it } != null) {
-      mSensorManager.registerListener(this, mSensor, SensorManager.SENSOR_DELAY_FASTEST)
+      mSensorManager.registerListener(this, mSensor, samplingPeriodUs)
     }
   }
 
   protected fun stopObserving() {
     mSensorManager.unregisterListener(this)
   }
+
+  private fun hasHighSamplingRateSensorsPermission(): Boolean {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S) {
+      return false
+    }
```

Comment:
Why return false by default?

I think this changes the previous default behaviour on older versions when we had always set `SENSOR_DELAY_FASTEST`. Now it'll be always `SENSOR_DELAY_NORMAL` on Android < S

If older Androids didn't need the manifest permission here (it was granted implicitly), I think we should return `true` here. But that's only my opinion 😉